server/functions.lua

```diff
@@ -83,3 +83,143 @@ function ToggleDoorState(entity)
         return 0
     end
 end
+
+local public = {}
+
+---Gets Citizen Id based on source
+---@param source number ID of the player
+---@return string? citizenid The player CitizenID, nil otherwise.
+local function getCitizenId(source)
+    local player = exports.qbx_core:GetPlayer(source)
+    if not player then return end
+
+    return player.PlayerData.citizenid
+end
+
+---Looking for a vehicle in the world
+---@param plate string The plate number of the vehicle.
+---@return number? vehicle The entity number of the found vehicle, nil otherwise.
+function public.findVehicleByPlate(plate)
+    local vehicles = GetAllVehicles()
+    for i = 1, #vehicles do
+        local vehicle = vehicles[i]
+        if qbx.getVehiclePlate(vehicle) == plate then
+            return vehicle
+        end
+    end
+end
+
+---Gives the player the vehicle keys item.
+---@param source number ID of the player
+---@param plate string The plate number of the vehicle.
+function public.giveKeysItem(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
+
+    local vehicleId = MySQL.single.await('SELECT id FROM player_vehicles WHERE citizenid = ? AND plate = ? LIMIT 1', { citizenid, plate })
+    exports.ox_inventory:AddItem(source, 'vehiclekeys', 1, { vehicleId = vehicleId, key = 'placeholder' })
+end
+
+---Takes the player the vehicle keys item.
+---@param source number ID of the player
+---@param plate string The plate number of the vehicle.
+function public.removeKeysItem(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
+
+    local vehicleId = MySQL.single.await('SELECT id FROM player_vehicles WHERE citizenid = ? AND plate = ? LIMIT 1', { citizenid, plate })
+    exports.ox_inventory:RemoveItem(source, 'vehiclekeys', 1, { vehicleId = vehicleId, key = 'placeholder' })
+end
```

Comment:
accidental key assignment by admin

Reply:
same cases as RemoveKeys function

Reply:
Items are a bit different, the user could go offline, but would still have the item. I think we should handle items by never removing it, instead just changing the locks on the car, so that the key item becomes useless.